### ISSUE TITLE
Shower of failures in tests with out '.' in @INC

### ISSUE THREAD

**Issue by kentfredric** [dagolden/Test-API#5]
On 5.25.10 with  -Ddefault_inc_excludes_dot 
```
PERL_DL_NONLAZY=1 "/home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/bin/perl5.25.10" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t

#   Failed test 'require t::lib::NoSubs;'
#   at t/03-export.t line 21.
#     Tried to require 't::lib::NoSubs'.
#     Error:  Can't locate t/lib/NoSubs.pm in @INC (you may need to install the t::lib::NoSubs module) (@INC contains: /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10 /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10) at (eval 6) line 2.

#   Failed test 'require t::lib::Export;'
#   at t/03-export.t line 22.
#     Tried to require 't::lib::Export'.
#     Error:  Can't locate t/lib/Export.pm in @INC (you may need to install the t::lib::Export module) (@INC contains: /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10 /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10) at (eval 7) line 2.

#   Failed test 'require t::lib::ExportComplex;'
#   at t/03-export.t line 23.
#     Tried to require 't::lib::ExportComplex'.
#     Error:  Can't locate t/lib/ExportComplex.pm in @INC (you may need to install the t::lib::ExportComplex module) (@INC contains: /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10 /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10) at (eval 8) line 2.

#   Failed test 'import_ok - export: none; expect: none'
#   at t/03-export.t line 41.
# STDOUT is:
# > not ok 1 - importing from t::lib::NoSubs
# not:
# > ok 1 - importing from t::lib::NoSubs
# as expected
# STDERR is:
# > #   Failed test 'importing from t::lib::NoSubs'
# > #   at t/03-export.t line 40.
# > # Module 't::lib::NoSubs' not loaded
# not:
# > 
# > 
# > 
# as expected

#   Failed test 'require t::lib::NoSubs;'
#   at t/02-public.t line 21.
#     Tried to require 't::lib::NoSubs'.
#     Error:  Can't locate t/lib/NoSubs.pm in @INC (you may need to install the t::lib::NoSubs module) (@INC contains: /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10 /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10) at (eval 6) line 2.

#   Failed test 'require t::lib::SubFoo;'
#   at t/02-public.t line 22.
#     Tried to require 't::lib::SubFoo'.
#     Error:  Can't locate t/lib/SubFoo.pm in @INC (you may need to install the t::lib::SubFoo module) (@INC contains: /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10 /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10) at (eval 7) line 2.

#   Failed test 'require t::lib::UseCarp;'
#   at t/02-public.t line 23.
#     Tried to require 't::lib::UseCarp'.
#     Error:  Can't locate t/lib/UseCarp.pm in @INC (you may need to install the t::lib::UseCarp module) (@INC contains: /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10 /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10) at (eval 8) line 2.

#   Failed test 'require t::lib::PvtFoo;'
#   at t/02-public.t line 24.
#     Tried to require 't::lib::PvtFoo'.
#     Error:  Can't locate t/lib/PvtFoo.pm in @INC (you may need to install the t::lib::PvtFoo module) (@INC contains: /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10 /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10) at (eval 9) line 2.

#   Failed test 'import_ok - export: none; expect: foo'
#   at t/03-export.t line 47.
# STDERR is:
# > #   Failed test 'importing from t::lib::NoSubs'
# > #   at t/03-export.t line 46.
# > # Module 't::lib::NoSubs' not loaded
# not:
# > /#\s+Failed\ test.*?\n?.*?at\ t\/03\-export\.t line 46.*\n?/
# > # not exported: foo
# > 
# as expected

#   Failed test 'import_ok - export: foo; expect: foo'
#   at t/03-export.t line 55.
# STDOUT is:
# > not ok 1 - importing from t::lib::Export
# not:
# > ok 1 - importing from t::lib::Export
# as expected
# STDERR is:
# > #   Failed test 'importing from t::lib::Export'
# > #   at t/03-export.t line 54.
# > # Module 't::lib::Export' not loaded
# not:
# > 
# > 
# > 
# as expected

#   Failed test 'public_ok - none provided, none expected'
#   at t/02-public.t line 34.
# STDOUT is:
# > not ok 1 - public API for t::lib::NoSubs
# not:
# > ok 1 - public API for t::lib::NoSubs
# as expected
# STDERR is:
# > #   Failed test 'public API for t::lib::NoSubs'
# > #   at t/02-public.t line 33.
# > # Module 't::lib::NoSubs' not loaded
# not:
# > 
# > 
# > 
# as expected

#   Failed test 'import_ok - export: foo; expect: none'
#   at t/03-export.t line 61.
# STDERR is:
# > #   Failed test 'importing from t::lib::Export'
# > #   at t/03-export.t line 60.
# > # Module 't::lib::Export' not loaded
# not:
# > /#\s+Failed\ test.*?\n?.*?at\ t\/03\-export\.t line 60.*\n?/
# > # unexpectedly exported: foo
# > 
# as expected

#   Failed test 'public_ok - none provided, one expected'
#   at t/02-public.t line 40.
# STDERR is:
# > #   Failed test 'public API for t::lib::NoSubs'
# > #   at t/02-public.t line 39.
# > # Module 't::lib::NoSubs' not loaded
# not:
# > /#\s+Failed\ test.*?\n?.*?at\ t\/02\-public\.t line 39.*\n?/
# > # missing: foo
# > 
# as expected

#   Failed test 'import_ok - export: several; expect: several'
#   at t/03-export.t line 73.
# STDOUT is:
# > not ok 1 - importing from t::lib::ExportComplex
# not:
# > ok 1 - importing from t::lib::ExportComplex
# as expected
# STDERR is:
# > #   Failed test 'importing from t::lib::ExportComplex'
# > #   at t/03-export.t line 68.
# > # Module 't::lib::ExportComplex' not loaded
# not:
# > 
# > 
# > 
# as expected

#   Failed test 'public_ok - one provided, none expected'
#   at t/02-public.t line 46.
# STDERR is:
# > #   Failed test 'public API for t::lib::SubFoo'
# > #   at t/02-public.t line 45.
# > # Module 't::lib::SubFoo' not loaded
# not:
# > /#\s+Failed\ test.*?\n?.*?at\ t\/02\-public\.t line 45.*\n?/
# > # extra: foo
# > 
# as expected

#   Failed test 'import_ok - export: several; expect: none'
#   at t/03-export.t line 83.
# STDERR is:
# > #   Failed test 'importing from t::lib::ExportComplex'
# > #   at t/03-export.t line 78.
# > # Module 't::lib::ExportComplex' not loaded
# not:
# > /#\s+Failed\ test.*?\n?.*?at\ t\/03\-export\.t line 78.*\n?/
# > # unexpectedly exported: bar foo
# > 
# as expected

#   Failed test 'public_ok - one provided, one expected'
#   at t/02-public.t line 50.
# STDOUT is:
# > not ok 1 - public API for t::lib::SubFoo
# not:
# > ok 1 - public API for t::lib::SubFoo
# as expected
# STDERR is:
# > #   Failed test 'public API for t::lib::SubFoo'
# > #   at t/02-public.t line 49.
# > # Module 't::lib::SubFoo' not loaded
# not:
# > 
# > 
# > 
# as expected

#   Failed test 'import_ok - export: several; expect: one'
#   at t/03-export.t line 93.
# STDERR is:
# > #   Failed test 'importing from t::lib::ExportComplex'
# > #   at t/03-export.t line 88.
# > # Module 't::lib::ExportComplex' not loaded
# not:
# > /#\s+Failed\ test.*?\n?.*?at\ t\/03\-export\.t line 88.*\n?/
# > # unexpectedly exported: bar
# > 
# as expected

#   Failed test 'public_ok - use Carp + missing sub'
#   at t/02-public.t line 56.
# STDERR is:
# > #   Failed test 'public API for t::lib::UseCarp'
# > #   at t/02-public.t line 55.
# > # Module 't::lib::UseCarp' not loaded
# not:
# > /#\s+Failed\ test.*?\n?.*?at\ t\/02\-public\.t line 55.*\n?/
# > # missing: foo
# > # extra: carp confess croak
# as expected

#   Failed test 'public_ok - private provided, none expected'
#   at t/02-public.t line 60.
# STDOUT is:
# > not ok 1 - public API for t::lib::PvtFoo
# not:
# > ok 1 - public API for t::lib::PvtFoo
# as expected
# STDERR is:
# > #   Failed test 'public API for t::lib::PvtFoo'
# > #   at t/02-public.t line 59.
# > # Module 't::lib::PvtFoo' not loaded
# not:
# > 
# > 
# > 
# as expected
t/02-public.t .......... 
Failed 10/11 subtests 

#   Failed test 'import_ok - export_ok several functions'
#   at t/03-export.t line 105.
# STDOUT is:
# > not ok 1 - importing from t::lib::ExportComplex
# not:
# > ok 1 - importing from t::lib::ExportComplex
# as expected
# STDERR is:
# > #   Failed test 'importing from t::lib::ExportComplex'
# > #   at t/03-export.t line 100.
# > # Module 't::lib::ExportComplex' not loaded
# not:
# > 
# > 
# > 
# as expected

#   Failed test 'import_ok - export_ok a missing function'
#   at t/03-export.t line 115.
# STDERR is:
# > #   Failed test 'importing from t::lib::ExportComplex'
# > #   at t/03-export.t line 110.
# > # Module 't::lib::ExportComplex' not loaded
# not:
# > /#\s+Failed\ test.*?\n?.*?at\ t\/03\-export\.t line 110.*\n?/
# > # not optionally exportable: wibble
# > 
# as expected

#   Failed test 'import_ok - export_ok excludes items in @EXPORT_OK'
#   at t/03-export.t line 121.
# STDERR is:
# > #   Failed test 'importing from t::lib::ExportComplex'
# > #   at t/03-export.t line 120.
# > # Module 't::lib::ExportComplex' not loaded
# not:
# > /#\s+Failed\ test.*?\n?.*?at\ t\/03\-export\.t line 120.*\n?/
# > # extra optionally exportable: bam baz
# > 
# as expected
t/03-export.t .......... 
Failed 13/14 subtests 

#   Failed test 'require t::lib::Parent;'
#   at t/04-class-ok.t line 21.
#     Tried to require 't::lib::Parent'.
#     Error:  Can't locate t/lib/Parent.pm in @INC (you may need to install the t::lib::Parent module) (@INC contains: /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10 /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10) at (eval 6) line 2.

#   Failed test 'require t::lib::Child;'
#   at t/04-class-ok.t line 22.
#     Tried to require 't::lib::Child'.
#     Error:  Can't locate t/lib/Child.pm in @INC (you may need to install the t::lib::Child module) (@INC contains: /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/lib /home/kent/.cpanm/work/1488182087.25118/Test-API-0.005/blib/arch /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/site_perl/5.25.10 /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10/x86_64-linux /home/kent/perl5/perlbrew/perls/5.25.10-nossp-sdbm-nopmc-nodot/lib/5.25.10) at (eval 7) line 2.

#   Failed test 'class_api_ok - missing method'
#   at t/04-class-ok.t line 32.
# STDERR is:
# > #   Failed test 'public API for class t::lib::Child'
# > #   at t/04-class-ok.t line 31.
# > # Module 't::lib::Child' not loaded
# not:
# > /#\s+Failed\ test.*?\n?.*?at\ t\/04\-class\-ok\.t line 31.*\n?/
# > # extra: www
# > 
# as expected

#   Failed test 'class_api_ok - complete set of methods'
#   at t/04-class-ok.t line 36.
# STDOUT is:
# > not ok 1 - public API for class t::lib::Child
# not:
# > ok 1 - public API for class t::lib::Child
# as expected
# STDERR is:
# > #   Failed test 'public API for class t::lib::Child'
# > #   at t/04-class-ok.t line 35.
# > # Module 't::lib::Child' not loaded
# not:
# > 
# > 
# > 
# as expected

#   Failed test 'class_api_ok - missing inherited method'
#   at t/04-class-ok.t line 40.
# STDOUT is:
# > not ok 1 - public API for class t::lib::Child
# not:
# > ok 1 - public API for class t::lib::Child
# as expected
# STDERR is:
# > #   Failed test 'public API for class t::lib::Child'
# > #   at t/04-class-ok.t line 39.
# > # Module 't::lib::Child' not loaded
# not:
# > 
# > 
# > 
# as expected
t/04-class-ok.t ........ 
Failed 5/5 subtests 
# 
# Versions for all modules listed in MYMETA.json (including optional ones):
#    Version Module                  
#   -------- ------------------------
#   2.150010 CPAN::Meta              
#      2.140 CPAN::Meta::Requirements
#       1.42 Carp                    
#       5.72 Exporter                
#       7.24 ExtUtils::MakeMaker     
#       3.66 File::Spec::Functions   
#    1.46_02 List::Util              
#       1.08 Symbol                  
#   1.302073 Test::Builder::Module   
#   1.302073 Test::Builder::Tester   
#   1.302073 Test::More              
#       1.11 strict                  
#     0.9917 version                 
#       1.37 warnings                
t/00-report-prereqs.t .. ok

Test Summary Report
-------------------
t/02-public.t        (Wstat: 0 Tests: 11 Failed: 10)
  Failed tests:  1-4, 6-11
t/03-export.t        (Wstat: 0 Tests: 14 Failed: 13)
  Failed tests:  1-3, 5-14
t/04-class-ok.t      (Wstat: 0 Tests: 5 Failed: 5)
  Failed tests:  1-5
Files=4, Tests=31,  0 wallclock secs ( 0.04 usr  0.01 sys +  0.46 cusr  0.08 csys =  0.59 CPU)
Result: FAIL
Failed 3/4 test programs. 28/31 subtests failed.
```

**Comment by xdg** [comment]
Shipped as 0.007-TRIAL.  Please confirm it works.

**Comment by kentfredric** [comment]
sending: (http://cpan.metacpan.org/authors/id/D/DA/DAGOLDEN/Test-API-0.007-TRIAL.tar.gz, DAGOLDEN, Test-API-0.007, PASS)
